saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -228,6 +228,10 @@ def _update_delivery_method(
     ) -> None:
         checkout_fields_to_update = ["shipping_method", "collection_point"]
         checkout = checkout_info.checkout
+        # Clear checkout shipping address if it was switched from C&C.
+        if checkout.collection_point_id and not collection_point:
```

Comment:
What if the collection point is changed to not CC with `ALL_WAREHOUSES` for click and collect option? We will end up with updated CC, but `shipping_address` for the previous `LOCAL_STOCK`. 
We should add a test for this case as well.